runtime/src/app/goto/index.ts

```diff
@@ -1,10 +1,10 @@
 import { history, select_target, navigate, cid } from '../app';
 
-export default function goto(href: string, opts = { replaceState: false }) {
+export default function goto(href: string, state = {}, opts = { replaceState: false }) {
```

Comment:
this would be a breaking change, which you could avoid by putting `state` in `opts`

Reply:
Yeah i know.. i didn't liked the idea of putting data inside a parameter called "options" and neither adding a third parameter knowing that most of the time it's likely you are gonne pushState and not replace it, resulting in to write `goto("path", null, myState)`.

But since you are right, at the moment we can move it under opts. Maybe in a next release the API could be enhanced.

Reply:
I'm not familiar with what the breaking change policy is. There's a few PRs that are flagged as breaking changes that will be merged when 0.28.0 is released. @antony or @Conduitry might have more details on that. But it would seem weird to me to have an `opts` parameter and not have the other parameters inside it because otherwise why not just have a boolean parameter for `replaceState` directly?

Reply:
I know, i fact i would likely created two different methods, pushState and replaceState as the standard API since i can't realize which other options we will ever see there.